File: backend/src/models/Commerce.js
const { Schema, model } = require('mongoose');
const bcrypt = require('bcryptjs');

const CommerceSchema = new Schema({
    name:        { type: String, required: true },
    email:       { type: String, required: true },
    password:    { type: String, required: true },
    category:    { type: String, require: true },
    country:     { type: String, require: true },
    state:       { type: String, require: true },
    description: { type: String, require: true,  default: null },
    reputation:  { type: Number, require: true, default: 0 }
});

CommerceSchema.methods.encryptPassword = (password) => {
    return bcrypt.hashSync(password, bcrypt.genSaltSync(10));
};

CommerceSchema.methods.validatePassword = function(password) {
    return bcrypt.compareSync(password, this.password);
};

module.exports = model('Commerce', CommerceSchema);

File: app_commerce/lib/Models/Commerce.dart
class Commerce {
  //Attributes
  final String id;
  final String email;
  final String name;
  final String country;
  final String state;
  final double reputation;
  final String category;
  final String description;
  final String imageURL;
  final String publicId;

  //Constructor
  Commerce({
    this.id,
    this.email,
    this.name,
    this.country,
    this.state,
    this.reputation,
    this.category,
    this.description,
    this.imageURL,
    this.publicId,
  });

  //Deserialize Json to List<Product>
  factory Commerce.fromJson(Map<String, dynamic> parsedJson) {
    return Commerce(
      id: parsedJson['_id'],
      name: parsedJson['name'],
      email: parsedJson['email'],
      description: parsedJson['description'],
      country: parsedJson['country'],
      state: parsedJson['state'],
      category: parsedJson['category'],
      imageURL: parsedJson['imageURL'],
      publicId: parsedJson['public_id'],
    );
  }
}
